### DISCUSSION TITLE
how to view multiple images

### DISCUSSION THREAD

**Discussion by dsb66** [jcorporation/myMPD#1314] [Q&A] [answered]
My album folders contain a front cover and a back cover file, following the setup described at the https://jcorporation.github.io/myMPD/references/pictures. In myMPD, I can see either the front cover image or the back cover image (after deleting the front cover), but I could not find a way two display or access both images. Can anyone recommend the correct settings to display both the cover and back images, either at the same time or sequentially? Based on the text at the link above, I assume this is possible.

**Comment by jcorporation** [comment]
Where would you see more than one image?

**Comment by dsb66** [comment]
I would like to know if it is possible to access/view more than one image. The instructions say that, in direct access mode, myMPD supports more than one image, e.g. cover, back, inlet, and I am assuming that all these multiple images would then be accessible through myMPD. 

For example, when viewing the song list, tags, and cover of one album, perhaps it may be possible to see the back cover next to the front cover, or cycle through all available images using a clickable arrow, or additional images may be accessible by download like the booklet is. 

I am a bit confused by the instructions and I would like to know if the behavior I am seeing is what should be expected or if I am doing something wrong with my setup. The suggestion that myMPD supports more than one image was very exciting.

**Reply by jcorporation** [reply]
If you click on a cover it should open a modal where you can cycle through the available images. This should work in the playback view, Cover in the footer or in the album detail view.

**Reply by dsb66** [reply]
I tried that, but after clicking on the cover I only get a larger view of the cover by no option to cycle through. I tried cycling either through keyboard arrows or clicking on the enlarged image.

**Reply by jcorporation** [reply]
There is no option to enable this. Let my try that in my setup.

**Reply by jcorporation** [reply, marked as answer]
It seems to be a bug, I open an issue to be fixed for the next release.

**Reply by dsb66** [reply]
I downloaded the most recent devel branch including the commit to fix this bug, but I am still unable to circle through cover and back images. After reinstalling myMPD, I also cleared the browser cache and rescanned the database, to no avail.

**Reply by jcorporation** [reply]
Sure myMPD has access to the music directory?

Please attach a har file from your browser. I must have a look at the responses from myMPD.

**Comment by dsb66** [comment]
Yes, myMPD can access the music directory. I can see the `back.jpg` in album view when that is the only image file in the album folder, and I can download the booklet (which I believe depends on accessing the music directory. I attach below the zipped har file while trying to view the album art (I am using MS Edge), and myMPD log showing connection through mpd's socket and detection of the music directory - thank you.

[albumart_test.zip](https://github.com/user-attachments/files/15945576/albumart_test.zip)

```
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  "stickerdb": Connected to MPD
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  "default": Connecting to socket "/run/mpd/socket"
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  "default": Connected to MPD
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  "default": MPD protocol version: 0.24.0
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  Directory: "/data"
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  Directory: "/var/lib/mpd/playlists"
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  Enabled tag_list: Artist Album AlbumArtist AlbumArtistSort Title Track Genre Date Composer Disc
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  Enabled tag_list_search: Album AlbumArtist Artist Composer Genre Title
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  Enabled tag_list_browse: Album AlbumArtist Artist Composer Genre Title
Jun 23 11:02:36 okto mympd[2841]: NOTICE   mympdapi  Enabled smartpls_generate_tag_list:
Jun 23 11:02:38 okto mympd[2841]: NOTICE   mympdapi  Starting mpd_worker thread for MYMPD_API_CACHES_CREATE
Jun 23 11:02:38 okto mympd[2841]: NOTICE   mpdworker "default": Connecting to socket "/run/mpd/socket"
Jun 23 11:02:38 okto mympd[2841]: NOTICE   mpdworker "default": Connected to MPD
Jun 23 11:02:38 okto mympd[2841]: NOTICE   mpdworker Stopping mpd_worker thread
Jun 23 11:03:06 okto mympd[2841]: NOTICE   mympdapi  Starting mpd_worker thread for MYMPD_API_SMARTPLS_UPDATE_ALL
Jun 23 11:03:06 okto mympd[2841]: NOTICE   mpdworker "default": Connecting to socket "/run/mpd/socket"
Jun 23 11:03:06 okto mympd[2841]: NOTICE   mpdworker "default": Connected to MPD
Jun 23 11:03:06 okto mympd[2841]: WARN     mpdworker $.value: JSON path "$.value" not found or value is not string type, found type is "MJSON_TOK_INVALID"
Jun 23 11:03:06 okto mympd[2841]: ERROR    mpdworker Can't parse smart playlist file "/var/lib/mympd/smartpls/myMPDsmart-mostPlayed" (sticker)
Jun 23 11:03:06 okto mympd[2841]: WARN     mpdworker $.value: JSON path "$.value" not found or value is not string type, found type is "MJSON_TOK_INVALID"
Jun 23 11:03:06 okto mympd[2841]: ERROR    mpdworker Can't parse smart playlist file "/var/lib/mympd/smartpls/myMPDsmart-bestRated" (sticker)
Jun 23 11:03:06 okto mympd[2841]: NOTICE   mpdworker 2 smart playlists updated, 1 already up-to-date
Jun 23 11:03:06 okto mympd[2841]: NOTICE   mpdworker Stopping mpd_worker thread
Jun 23 11:03:36 okto mympd[2841]: NOTICE   mympdapi  Starting mpd_worker thread for MYMPD_API_CACHE_DISK_CROP
Jun 23 11:03:36 okto mympd[2841]: NOTICE   mpdworker Deleted 0 files from cover cache
Jun 23 11:03:36 okto mympd[2841]: NOTICE   mpdworker Deleted 0 files from lyrics cache
Jun 23 11:03:36 okto mympd[2841]: NOTICE   mpdworker Deleted 0 files from thumbs cache
Jun 23 11:03:36 okto mympd[2841]: NOTICE   mpdworker Deleted 0 files from misc cache
Jun 23 11:03:36 okto mympd[2841]: NOTICE   mpdworker Stopping mpd_worker thread
```

**Reply by jcorporation** [reply]
You configured the coverimage names as `cover, back`. This is the problem, remove the `back`.

**Comment by dsb66** [comment]
Great, it is working now. I was confused by the instructions (_"Configure the name in the settings (albumart filenames and thumbnail names), a comma separated list of filenames and only basenames (filenames without extensions) are also supported"_), which indicate that all the filenames should be listed.
Thank you very much!
